### PR TITLE
Support for alloy_primitive int types

### PR DESCRIPTION
**Pull request by rory-ocl**
## Description

Support for alloy_primitives::Signed and alloy_primitives::Uint in return types and parameters.

## TODO

Still in draft as I finish up the following tasks:

- [x] Test all types in both return types and parameters
- [x] Ensure this does not increase gas cost for already-supported types and apply optimization if necessary

## Checklist

- [x] I have documented these changes where necessary.
- [x] I have read the [DCO][DCO] and ensured that these changes comply.
- [x] I assign this work under its [open source licensing][terms].

[DCO]: https://github.com/OffchainLabs/stylus-sdk-rs/blob/stylus/licenses/DCO.txt
[terms]: https://github.com/OffchainLabs/stylus-sdk-rs/blob/stylus/licenses/COPYRIGHT.md

### CODE REVIEW

File: stylus-sdk/src/abi/ints.rs [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+/// Represents [`intX`] in Solidity
+///
+/// A custom type is used here in lieu of [alloy_sol_types::sol_data::Int] in order to keep our
+/// bijectivity constraint.
+///
+/// [`intX`]: https://docs.soliditylang.org/en/latest/types.html#integers
+pub struct SolInt<const BITS: usize, const LIMBS: usize>;
+
+impl<const BITS: usize, const LIMBS: usize> AbiType for Signed<BITS, LIMBS>
+where
```
Comment (rauljordan) [inline stylus-sdk/src/abi/ints.rs:33]: This is the most hardcore generic constraint I've seen 😅, great job on getting it to work

File: stylus-sdk/src/abi/ints.rs [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        let slice = int.as_limbs();
+        if slice.len() < TL {
+            let mut limbs = [0; TL];
+            limbs[..slice.len()].copy_from_slice(slice);
+            Converted(Signed::from_limbs(limbs))
+        } else {
+            let (head, _tail) = slice.split_at(TL);
+            let mut limbs = [0; TL];
+            limbs.copy_from_slice(head);
+            /* TODO overflow check needed?
```
Comment (rauljordan) [inline stylus-sdk/src/abi/ints.rs:212]: Makes sense to be extra careful IMO

**Review by rauljordan — COMMENTED** [review]
LGTM just pending the overflow check

**Review by rauljordan — DISMISSED** [review]

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Pull request merged**